sktime/transformations/base.py

```diff
@@ -48,93 +104,567 @@ class BaseTransformer(BaseEstimator):
+        if X_was_Series and output_scitype == "Series":
+            Xt = convert_Panel_to_Series(Xt)
+
+        if output_scitype == "Series":
+            Xt = convert_to(
+                Xt,
+                to_type=X_input_mtype,
+                as_scitype=X_input_scitype,
+            )
+        elif output_scitype == "Primitives":
+            # we "abuse" the Series converter to ensure df output
+            Xt = convert_to(
+                Xt,
+                to_type="pd.DataFrame",
+                as_scitype="Series",
+            )
+        else:
+            # output_scitype is "Panel" and no need for conversion
+            pass
+
+        return Xt
 
-    def fit_transform(self, Z, X=None):
+    def fit_transform(self, X, y=None, Z=None):
         """Fit to data, then transform it.
 
         Fits transformer to X and y with optional parameters fit_params
         and returns a transformed version of X.
 
+        State change:
+            Changes state to "fitted".
+
+        Writes to self:
+            Sets is_fitted flag to True.
+            Sets fitted model attributes ending in "_".
+
         Parameters
         ----------
-        Z : pd.DataFrame, pd.Series or np.ndarray
-            Data to be transformed
-        X : pd.Series or np.ndarray, optional (default=None)
-            Target values of data to be transformed.
+        X : Series or Panel, any supported mtype
+            Data to be transformed, of python type as follows:
+                Series: pd.Series, pd.DataFrame, or np.ndarray (1D or 2D)
+                Panel: pd.DataFrame with 2-level MultiIndex, list of pd.DataFrame,
+                    nested pd.DataFrame, or pd.DataFrame in long/wide format
+                subject to sktime mtype format specifications, for further details see
+                    examples/AA_datatypes_and_datasets.ipynb
+        y : Series or Panel, optional (default=None)
+            Additional data, e.g., labels for transformation
+        Z : possible alias for X; should not be passed when X is passed
+            alias Z will be deprecated in version 0.10.0
 
         Returns
         -------
-        Zt : pd.DataFrame, pd.Series or np.ndarray
-            Transformed data.
+        transformed version of X
+        type depends on type of X and scitype:transform-output tag:
+            |   `X`    | `tf-output`  |     type of return     |
+            |----------|--------------|------------------------|
+            | `Series` | `Primitives` | `pd.DataFrame` (1-row) |
+            | `Panel`  | `Primitives` | `pd.DataFrame`         |
+            | `Series` | `Series`     | `Series`               |
+            | `Panel`  | `Series`     | `Panel`                |
+            | `Series` | `Panel`      | `Panel`                |
+        instances in return correspond to instances in `X`
+        combinations not in the table are currently not supported
         """
+        X = _handle_alias(X, Z)
```

Comment:
Do you need this when the first 2 lines of fit and transform call this method?

Reply:
Better safe than sorry? It's slightly easier to debug if it doesn't enter `fit`, and it's going to be removed eventually.